File: python/hot100/41.二叉树的层序遍历.py
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:return []
        q=deque([root])
        ans=[]
        while q:
            level=[]
            for _ in range(len(q)):
                cur=q.popleft()
                level.append(cur.val)
                if(cur.left): q.append(cur.left)
                if(cur.right): q.append(cur.right)
            ans.append(level)
        return ans

File: cpp/hot100/41.二叉树的层序遍历.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        queue<TreeNode*> q;
        vector<vector<int>> res;
        if(root) q.push(root);
        while(!q.empty()){
            vector<int> tmp;
            for(int i=q.size();i!=0;--i){
                root=q.front();
                q.pop();
                tmp.push_back(root->val);
                if(root->left) q.push(root->left);
                if(root->right) q.push(root->right);
            }
            res.push_back(tmp);
        }
        return res;
    }
};